### ISSUE TITLE
Pull an Image from a Private Registry

### ISSUE THREAD

**Issue by Karmavil** [kubernetes/website#26673]
/question

Hi guys.
I had a terrible time thanks to the fact that the documentation describes the use of the `email` and `auth` trying to [Create a Secret by providing credentials on the command line](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line).

I'm using `pass` in my docker config.json so [encoding the config.json](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials) file just doesn't work  in this case.

**Everything works fine just encoding the registry, name and password** with docker hub because it doesn't use email to login:
 ```{"auths": {"https://index.docker.io/v1/": {"username":"my-name","password":"my-password"}}}```

I didn't find too many questions about this so the documentation is probably fine but I'm sure I'm not the only one using `pass`
___
Docker hub make use of your DockerID (username) and password to login but you have to associate your email to create an account so it was difficult to understand what was wrong. **It's not the only registry out there so I understand the documentation is wright but I think that docker hub may require a special mention**.

Thanks for your time
Oh by the way I'm using minikube

**Comment by sftim** [comment]
Hi @Karmavil 

It looks to me that part of this story is covered in https://kubernetes.io/docs/concepts/containers/images/#using-a-private-registry and part in https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line

Do you think you have enough insight into what's not right with the current page to describe the shortcomings? If so, that will let a contributor pick up the work and propose a fix.

**Comment by Karmavil** [comment]
Hi @sftim 
I'm collecting information. 
Meaning that I didn't try authenticate yet using the auth and the real email as credentials, but I can tell for sure that I cannot authenticate using auth an "email"="none"
I have some extra info in a reply to another [issue related to minikube](https://github.com/kubernetes/minikube/issues/9161).
I would like to know what they think about it because they had an addon (registry-creds) that falls on this issue.  But it makes me think that there was a reason for that. 
Maybe docker hub changed something (?) I don't know. 
I'm waiting for their feedback because I couldn't find which group is the owner of kubectl or kubeconfig to ask them about this

**Comment by Karmavil** [comment]
I should update the description of this issue as soon as possible

**Comment by sftim** [comment]
This issue is about the part of Kubernetes that runs Pods (that is: nodes). SIG Node is the most relevant SIG here.

**Comment by Karmavil** [comment]
It's time to close this issue, I just checked and it's working.
I can authenticate with hub.docker and pull images without complains.

There was an update in the Kubernetes website [documentation page](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-by-providing-credentials-on-the-command-line): 18c22d133e573d85700cdfb7b5e74d130a654ea0 (Feb 16, 22days ago)
so I checked with the new url: https://index.docker.io/v2/
and the previous one too: https://index.docker.io/v1

and it works with both.. and this is the weird thing: I tried to authenticate using the console/terminal and it failed.. 
```
Authenticating with existing credentials...
** Message: Remote error from secret service: org.freedesktop.DBus.Error.Failed: Couldn't create item: The secret was transferred or encrypted in an invalid way.
Error saving credentials: error storing credentials - err: exit status 1, out: `Couldn't create item: The secret was transferred or encrypted in an invalid way.`
```
After manually update the config.json file to v2..  Login Succeeded
( now my config.json has 2 auths so they won't catch me again :D )

I opened this Issue because a tiny possibility for the need of the documentation update 
but now I see that this problems belongs to the [repository and its documentation](https://docs.docker.com/engine/reference/commandline/login/).
It was obvious it's beyond the kubernetes documentation website..
